src/fsharp/fsi/console.fs

```diff
@@ -17,7 +17,7 @@ module internal ConsoleOptions =
   let fixupRequired = false
 #else
   // Bug 4254 was fixed in Dev11 (Net4.5), so this flag tracks making this fix up version specific.
-  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove
+  let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && Type.GetType("Mono.Runtime") = null
```

Comment:
OK, I see. So I think the condition should never run when Mono is running, so just use this?

```
let fixupRequired = not FSharpEnvironment.IsRunningOnNetFx45OrAbove && not runningOnMono
```

But please verify by hand that this solves the problem.